### PR TITLE
Add `euclidan_surface_jacobian`

### PR DESCRIPTION
**Pull request by nikwit**
Adds a function that computes the euclidean surface element from the inverse jacobian.

~~depends on #4719~~

### CODE REVIEW

File: src/Domain/SurfaceElement.hpp
Comment (nilsvu) [inline src/Domain/SurfaceElement.hpp]: Please format equations like in `Domain/Tags/SurfaceJacobian.hpp`

File: src/Domain/SurfaceElement.hpp
Comment (nilsvu) [inline src/Domain/SurfaceElement.hpp]: Don't you need both curved space and time dependence for the worldtube integrals?

Also, you actually have to evaluate this at every time step, right? In that case it might be necessary for performance to split the computation into ElementLogical -> Grid and Grid -> Inertial, so you don't have to evaluate the static ElementLogical -> Grid at every time step. Up to you if you want to do this now or add notes to the docs that it's a possible performance optimization.

File: src/Domain/SurfaceElement.hpp
Comment (nilsvu) [inline src/Domain/SurfaceElement.hpp]: I suggest you name these things `SurfaceJacobian` to avoid name collisions with "element". There's already a tag in `Domain/Tags/SurfaceJacobian.hpp` with that name.

File: src/Domain/SurfaceElement.hpp
Comment (nilsvu) [inline src/Domain/SurfaceElement.hpp]: \param in new line

File: src/Domain/SurfaceElement.cpp
Comment (nilsvu) [inline src/Domain/SurfaceElement.cpp]: delete

File: src/Domain/SurfaceElement.cpp
Comment (nilsvu) [inline src/Domain/SurfaceElement.cpp]: include ErrorHandling/Assert

Comment (nilsvu) [inline src/Domain/SurfaceElement.cpp]: prefer `get<0, 0>(tensor)` when you know the indices at compile time. Same for `get(scalar)` over `scalar.get()`.

File: src/Domain/SurfaceElement.cpp
Comment (nilsvu) [inline src/Domain/SurfaceElement.cpp]: delete

File: src/Domain/SurfaceElement.hpp
Comment (nilsvu) [inline src/Domain/SurfaceElement.hpp]: delete

File: tests/Unit/Domain/Test_SurfaceElement.cpp
Comment (nilsvu) [inline tests/Unit/Domain/Test_SurfaceElement.cpp]: reminder to prefer `get<0>(coords)` over `coords.get(0)` when you know the indices at compile time

File: tests/Unit/Domain/Test_SurfaceElement.cpp
Comment (nilsvu) [inline tests/Unit/Domain/Test_SurfaceElement.cpp]: this could be `for (const auto& [inner_radius, ref_level] : cartesian_product(make_array(1.e-4, 1.e-2, 1.4), make_array(0, 1, 2))` to flatten the loop

**Review by nilsvu — CHANGES_REQUESTED** [review]
nice tests!

File: src/Domain/SurfaceElement.hpp
Comment (nikwit) [reply src/Domain/SurfaceElement.hpp]: No, I don't need those for now. The worldtube currently  represents the internal solution inside the worldtube in the grid frame and uses a euclidean coordinates.

I might need to add them later for non-circular orbits. The overloads should probably be the same ones as for `unnormalized_normal_vector`.

File: src/Domain/SurfaceElement.hpp
Comment (nikwit) [reply src/Domain/SurfaceElement.hpp]: oops, had the wrong formatter config

File: src/Domain/SurfaceElement.hpp
Comment (nilsvu) [reply src/Domain/SurfaceElement.hpp]: Would you mind adding an overload that takes a spatial metric? We'll need that sooner or later. Do you currently get by with the Euclidean surface integral because you're working with a perturbative solution in the worldtube? Would it be wrong to use the spatial metric in the integral?

**Comment by nilsvu** [comment]
Fixup LGTM, please squash and rebase once the dependency PR is merged.

File: src/Domain/SurfaceElement.hpp
Comment (nikwit) [reply src/Domain/SurfaceElement.hpp]: Yes, the local solution is in Cartesian coordinates. I can add the proper curved space integral in a follow-up PR. Do you happen to know about a nice test  for that  like an analytical solution for a scalar quantity integrated over a sphere centered on a Schwarzschild/Kerr black hole?

Comment (nilsvu) [reply src/Domain/SurfaceElement.hpp]: Yea the irreducible mass (horizon area) of a Kerr horizon is a good quantity to compute [Eq. (1.70) in Baumgarte/Shapiro]. It's also computed for a Strahlkorper here: https://github.com/sxs-collaboration/spectre/blob/develop/tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp#L928

File: src/Domain/SurfaceElement.hpp
Comment (nilsvu) [reply src/Domain/SurfaceElement.hpp]: Also name the functions `euclidean_surface_jacobian` please

File: src/Domain/SurfaceJacobian.hpp
Comment (nilsvu) [inline src/Domain/SurfaceJacobian.hpp]: Also name the functions `euclidean_surface_jacobian` please

**Review by nilsvu — CHANGES_REQUESTED** [review]
squash directly

File: src/Domain/SurfaceJacobian.hpp
Comment (nikwit) [reply src/Domain/SurfaceJacobian.hpp]: I think this is more consistent with `euclidean_area_element` in `ApparentHorizons/StrahlKorperGr.hpp` which calculates the same thing for Strahlkorpers.

So I guess just `euclidean_area_element` or do you prefer `euclidean_surface_jacobian`

Comment (nilsvu) [reply src/Domain/SurfaceJacobian.hpp]: Either works, just make the tag, the functions, and the file names consistent. I suggested "surface jacobian" just because that's the name of the existing tag.

**Review by nilsvu — APPROVED** [review]

**Pull request merged**
